pkg/server/server.go

```diff
@@ -619,14 +615,17 @@ func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openf
 	}
 
 	ctx = typesystem.ContextWithTypesystem(ctx, typesys)
-
-	checkResolver := graph.NewLocalChecker(
-		storagewrappers.NewCombinedTupleReader(s.datastore, req.ContextualTuples.GetTupleKeys()),
-		s.checkOptions...,
+	ctx = storage.ContextWithRelationshipTupleReader(ctx,
+		storagewrappers.NewBoundedConcurrencyTupleReader(
+			storagewrappers.NewCombinedTupleReader(
```

Comment:
Following what I said in https://github.com/openfga/openfga/pull/1335#discussion_r1480652748, if this is the order of instantation done in `main` today too, then it's incorrect. 

When you call `Read` on a `b=boundedConcurrencyTupleReader` you are effectively doing this:

```go
       b.waitForLimiter(ctx)

	defer func() {
		<-b.limiter
	}()

	return b.RelationshipTupleReader.ReadUserTuple(ctx, store, tupleKey)
```

which is:

```go
        b.waitForLimiter(ctx)

	defer func() {
		<-b.limiter
	}()

	iter1 := storage.NewStaticTupleIterator(filterTuples(c.contextualTuples, tk.Object, tk.Relation))

	iter2, err := c.RelationshipTupleReader.Read(ctx, storeID, tk)
	if err != nil {
		return nil, err
	}

	return storage.NewCombinedIterator(iter1, iter2), nil
```

so I think in practice we are rate-limiting reading of things that are in memory.

The composition of functions should be this

```
storagewrappers.NewCombinedTupleReader(
			storagewrappers.NewBoundedConcurrencyTupleReader(
```